File: pet-clinic-data/src/main/java/com/sachet/sfgpetclinic/services/PetService.java
package com.sachet.sfgpetclinic.services;

import com.sachet.sfgpetclinic.model.Pet;

import java.util.Set;

public interface PetService {

    Pet findById(Long id);

    Pet save(Pet pet);

    Set<Pet> findAll();

    void delete(Pet pet);

    void deleteById(Long id);

}

File: src/main/kotlin/com/sachet/sfgpetclinic/service/PetService.kt
package com.sachet.sfgpetclinic.service

import com.sachet.sfgpetclinic.model.Pet

interface PetService {

    fun findById(id: Long): Pet
    fun save(pet: Pet): Pet
    fun findAll(): Set<Pet>
    fun delete(pet: Pet)


}